tests/unit/preprocessor/test_preprocessor_file.py

```diff
@@ -0,0 +1,142 @@
+"""Unit tests for :class:`esmvalcore.preprocessor.PreprocessorFile`."""
+
+from unittest import mock
+
+import pytest
+from iris.cube import Cube, CubeList
+
+from esmvalcore.preprocessor import PreprocessorFile
+
+ATTRIBUTES = {
+    'filename': 'file.nc',
+    'standard_name': 'precipitation',
+    'long_name': 'Precipitation',
+    'short_name': 'pr',
+    'units': 'kg m-2 s-1',
+    'frequency': 'mon',
+}
+
+
+@pytest.fixture
+def product():
+    """PreprocessorFile object used for testing."""
+    cube = Cube(
+        0,
+        var_name='tas',
+        standard_name='air_temperature',
+        long_name='Near-Surface Air Temperature',
+        units='K',
+        attributes={'frequency': 'day'},
+    )
+    product = PreprocessorFile(attributes=ATTRIBUTES, settings={})
+    product._cubes = CubeList([cube, cube, cube])
+    return product
+
+
+def test_update_attributes_empty_cubes(product):
+    """Test ``_update_attributes``."""
+    product._cubes = CubeList([])
+    product._update_attributes()
+
+    assert not product._cubes
+    assert product.attributes == ATTRIBUTES
+
+
+def test_update_attributes(product):
+    """Test ``_update_attributes``."""
+    product._update_attributes()
+
+    assert product.attributes == {
+        'filename': 'file.nc',
+        'standard_name': 'air_temperature',
+        'long_name': 'Near-Surface Air Temperature',
+        'short_name': 'tas',
+        'units': 'K',
+        'frequency': 'day',
+    }
```

Comment:
we should start thinking using dedicated tools that do comparisons like [deepdiff](https://pypi.org/project/deepdiff/)

Reply:
Looks definitely interesting, but maybe a bit of an overkill here (since it's only a flat dict)?

Reply:
yeah not here, more like a me being trying to be a showoff remark :laughing: Just a pointer for future, on a serious note